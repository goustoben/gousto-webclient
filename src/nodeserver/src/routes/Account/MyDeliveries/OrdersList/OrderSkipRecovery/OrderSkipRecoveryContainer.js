import { connect } from 'react-redux'

import { keepOrder, cancelPendingOrder } from 'actions/orderSkipRecovery'

import OrderSkipRecovery from './OrderSkipRecovery'

const mapStateToProps = (state) => ({
    visible: state.orderSkipRecovery.get('modalVisibility'),
    orderId: state.orderSkipRecovery.get('orderId'),
})

const mapDispatchToProps = {
    keepOrder,
    cancelOrder: cancelPendingOrder,
}

const OrderSkipRecoveryContainer = connect(mapStateToProps, mapDispatchToProps)(OrderSkipRecovery)

export default OrderSkipRecoveryContainer
