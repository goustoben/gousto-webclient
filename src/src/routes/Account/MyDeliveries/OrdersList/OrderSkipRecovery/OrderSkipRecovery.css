@value Bluecheese, White, Black from 'styles/colors.css';
@value comfortaaFontStack, avenirHeavyFontStack from 'styles/typography.css';

.keepButton {
    display: block;
    height: 50px;
    width: 250px;
    border: 0.75px solid Bluecheese;
    border-radius: 25px;
    background-color: Bluecheese;
    margin: 23px auto;
    font-family: avenirHeavyFontStack;
    text-align: center;
    line-height: 22px;
    font-size: 16px;
    font-weight: 900;
    color: White;
}

.skipAnyWay {
    display: block;
    color: Bluecheese;
    font-family: Avenir;
    font-size: 16px;
    font-weight: 900;
    line-height: 19px;
    text-align: center;
    margin: 30px auto 10px;
    cursor: pointer;
}

.valuePropositionTitle {
    color: Black;
    font-family: Avenir;
    font-size: 16px;
    font-weight: 500;
    line-height: 16px;
    text-align: center;
    margin: 38px auto 10px;
}

.valuePropositionDescription{
    color: Black;
    font-family: Avenir;
    font-size: 14px;
    font-weight: 300;
    line-height: 19px;
    text-align: center;
}
