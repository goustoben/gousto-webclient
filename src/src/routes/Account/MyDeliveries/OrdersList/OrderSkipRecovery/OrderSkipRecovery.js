import React, { PropTypes } from 'react'

import ModalComponent from 'ModalComponent'

import Title from './Title'
import ValueProposition from './ValueProposition'
import Footer from './Footer'

const propTypes = {
	visible: PropTypes.bool,
	orderId: PropTypes.string,
	dayId: PropTypes.string,
	orderType: PropTypes.string,
	featureFlag: PropTypes.bool,
	keepOrder: PropTypes.func.isRequired,
	cancelPendingOrder: PropTypes.func.isRequired,
	cancelProjectedOrder: PropTypes.func.isRequired,
	title: PropTypes.string,
	valueProposition: PropTypes.shape({
		message: PropTypes.string,
		title: PropTypes.string,
	}),
	callToActions: PropTypes.shape({
		confirm: PropTypes.string,
		keep: PropTypes.string,
	})
}

class OrderSkipRecovery extends React.PureComponent {

	componentDidUpdate(prevProps) {
		const { triggered, orderId, orderDate, dayId, orderType, getSkipRecoveryContent } = this.props
		const actionTriggered = (orderType === 'pending') ? 'Cancel' : 'Skip'

		if (triggered && (prevProps.triggered !== triggered)) {

			getSkipRecoveryContent({
				orderId,
				orderDate,
				dayId,
				orderType,
				actionTriggered,
			})
		}
	}

	skipCancelOrder(orderId, dayId, orderType, cancelPendingOrder, cancelProjectedOrder) {
		if (orderType === 'pending') {
			cancelPendingOrder(orderId)
		} else {
			cancelProjectedOrder(dayId)
		}
	}

	render() {
		const { visible, dayId, orderId, orderType, featureFlag, keepOrder, cancelPendingOrder, cancelProjectedOrder, title, valueProposition, callToActions } = this.props
		const onClickKeepOrder = () => keepOrder({ orderId, status: orderType })
		const onClickSkipCancel = () => this.skipCancelOrder(orderId, dayId, orderType, cancelPendingOrder, cancelProjectedOrder)

		return (
			<ModalComponent visible={visible}>
				<Title title={title} orderType={orderType} />
				<ValueProposition featureFlag={featureFlag} valueProposition={valueProposition} />
				<Footer orderType={orderType} callToActions={callToActions} onClickKeepOrder={onClickKeepOrder} onClickSkipCancel={onClickSkipCancel} />
			</ModalComponent>
		)
	}
}

OrderSkipRecovery.propTypes = propTypes

export { OrderSkipRecovery }
