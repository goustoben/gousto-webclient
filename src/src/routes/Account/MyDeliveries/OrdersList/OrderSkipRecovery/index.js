export { OrderSkipRecoveryContainer as OrderSkipRecovery } from './OrderSkipRecoveryContainer'
