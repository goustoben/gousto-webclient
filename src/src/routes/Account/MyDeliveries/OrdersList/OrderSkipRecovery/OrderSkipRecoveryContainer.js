import { connect } from 'react-redux'

import { getSkipRecoveryContent, keepOrder, cancelPendingOrder, cancelProjectedOrder } from 'actions/orderSkipRecovery'

import { OrderSkipRecovery } from './OrderSkipRecovery'

const mapStateToProps = (state) => ({
    triggered: state.orderSkipRecovery.get('triggered'),
    visible: state.orderSkipRecovery.get('modalVisibility'),
    orderId: state.orderSkipRecovery.get('orderId'),
    orderDate: state.orderSkipRecovery.get('orderDate'),
    dayId: state.orderSkipRecovery.get('dayId'),
    orderType: state.orderSkipRecovery.get('orderType'),
    boxNumber: state.subscription.getIn(['subscription', 'currentBoxNumber']),
    featureFlag: state.features.getIn(['skipRecovery', 'value']),
    title: state.orderSkipRecovery.get('title'),
    valueProposition: state.orderSkipRecovery.get('valueProposition'),
    callToActions: state.orderSkipRecovery.get('callToActions'),
})

const mapDispatchToProps = {
    getSkipRecoveryContent,
    keepOrder,
    cancelPendingOrder,
    cancelProjectedOrder,

}

export const OrderSkipRecoveryContainer = connect(mapStateToProps, mapDispatchToProps)(OrderSkipRecovery)
