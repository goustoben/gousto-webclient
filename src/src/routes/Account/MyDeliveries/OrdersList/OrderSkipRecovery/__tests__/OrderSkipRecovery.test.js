import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { OrderSkipRecovery } from '../OrderSkipRecovery'

jest.mock('components/Overlay', () => 'Overlay') // don't want to be testinng this

jest.mock('../Title', () => 'Title')
jest.mock('../Offer', () => 'Offer')
jest.mock('../ValueProposition', () => 'ValueProposition')
jest.mock('../Footer', () => 'Footer')

describe('Order Skip Recovery Modal', () => {
  let wrapper
  const keepOrder = jest.fn()
  const cancelPendingOrder = jest.fn()
  const cancelProjectedOrder = jest.fn()

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('Initial Render', () => {
    const valueProposition = {
      title: 'value proposition title',
      message: 'value proposition message',
    }
    const callToActions = {
      keep: 'keep',
      confirm: 'confirm',
    }

    beforeAll(() => {
      wrapper = shallow(
				<OrderSkipRecovery
				  title="modal title"
				  orderType="pending"
				  keepOrder={keepOrder}
				  cancelPendingOrder={cancelPendingOrder}
				  cancelProjectedOrder={cancelProjectedOrder}
				  featureFlag
				  valueProposition={valueProposition}
				  callToActions={callToActions}
				/>
      )
    })

    test('should render snapshot', () => {
      const tree = renderer.create(
				<OrderSkipRecovery
				  featureFlag
				  keepOrder={keepOrder}
				  cancelPendingOrder={cancelPendingOrder}
				  cancelProjectedOrder={cancelProjectedOrder}
				/>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('should display modal title', () => {
      const modalTitle = wrapper.find('Title')

      expect(modalTitle.length).toBe(1)
    })

    test('should pass correct props to title', () => {
      const modalTitle = wrapper.find('Title')

      expect(modalTitle.props().title).toBe('modal title')
      expect(modalTitle.props().orderType).toBe('pending')
    })

    test('should display modal value proposition', () => {
      const modalValueProposition = wrapper.find('ValueProposition')

      expect(modalValueProposition.length).toBe(1)
    })

    test('should pass correct props to value proposition', () => {
      const modalValueProposition = wrapper.find('ValueProposition')

      expect(modalValueProposition.props().valueProposition).toEqual(valueProposition)
    })

    test('should display modal footer', () => {
      const footer = wrapper.find('Footer')

      expect(footer.length).toBe(1)
    })

    test('should pass correct props to footer', () => {
      const footer = wrapper.find('Footer')

      expect(footer.props().orderType).toBe('pending')
      expect(footer.props().callToActions).toEqual(callToActions)
      expect(typeof footer.props().onClickKeepOrder).toBe('function')
      expect(typeof footer.props().onClickSkipCancel).toBe('function')
    })
  })

  describe('Alternative Render', () => {
    test('should not render modal content if featureflag is false', () => {
      wrapper = shallow(
				<OrderSkipRecovery
				  keepOrder={keepOrder}
				  cancelPendingOrder={cancelPendingOrder}
				  cancelProjectedOrder={cancelProjectedOrder}
				/>
      )

      const modalValueProposition = wrapper.find('ValueProposition')

      expect(modalValueProposition.length).toBe(0)
    })

    test('should only call cancel pending order when order type is pending', () => {
      wrapper = shallow(
				<OrderSkipRecovery
				  keepOrder={keepOrder}
				  cancelPendingOrder={cancelPendingOrder}
				  cancelProjectedOrder={cancelProjectedOrder}
				/>
      )

      wrapper.instance().skipCancelOrder('13123', '123123', 'pending', cancelPendingOrder, cancelProjectedOrder)

      expect(cancelProjectedOrder).toHaveBeenCalledTimes(0)
      expect(cancelPendingOrder).toHaveBeenCalledTimes(1)
    })

    test('should only call cancel projected order when order type is projected', () => {
      wrapper = shallow(
				<OrderSkipRecovery
				  keepOrder={keepOrder}
				  cancelPendingOrder={cancelPendingOrder}
				  cancelProjectedOrder={cancelProjectedOrder}
				/>
      )

      wrapper.instance().skipCancelOrder('13123', '123123', 'projected', cancelPendingOrder, cancelProjectedOrder)

      expect(cancelPendingOrder).toHaveBeenCalledTimes(0)
      expect(cancelProjectedOrder).toHaveBeenCalledTimes(1)
    })

    test('should call getSkipRecoveryContent with appropriate props', () => {
      const getSkipRecoveryContent = jest.fn()
      wrapper = shallow(
				<OrderSkipRecovery
				  orderId="14245"
				  triggered={false}
				  orderDate="2018-09-24T13:27:09.487Z"
				  dayId="23001"
				  orderType="pending"
				  getSkipRecoveryContent={getSkipRecoveryContent}
				/>
      )

      const prevProps = wrapper.props()

      wrapper.setProps({
        triggered: true,
      })

      wrapper.instance().componentDidUpdate(prevProps)

      expect(getSkipRecoveryContent).toHaveBeenCalledWith({
        actionTriggered: 'Cancel',
        dayId: '23001',
        orderDate: '2018-09-24T13:27:09.487Z',
        orderId: '14245',
        status: 'pending',
      })
    })
  })
})
