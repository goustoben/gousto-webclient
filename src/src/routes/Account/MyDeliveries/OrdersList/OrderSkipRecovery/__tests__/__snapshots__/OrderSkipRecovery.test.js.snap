// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Recovery Modal Initial Render should render snapshot 1`] = `
<Overlay
  from="top"
  open={false}
>
  <div
    className="modalComponent"
  >
    <div
      className="spacer"
    />
    <div
      className="container"
    >
      <div
        className="modalTitle"
      >
        <Title />
      </div>
      <Footer
        onClickKeepOrder={[Function]}
        onClickSkipCancel={[Function]}
      />
    </div>
  </div>
</Overlay>
`;
